### ISSUE TITLE
`Style/ConditionalAssignment` is incorrectly flagging use of `===` in `case` statement

### ISSUE THREAD

**Issue by postmodern** [rubocop/rubocop#12686]
I noticed that rubocop was flagging this code as a violation of `Style/ConditionalAssignment`, despite it not performing any assignment.

```ruby
def ===(other)
  case other
  when IPRange
    @range === other.range
  else
    @range === other
  end
end
```

Running `rubocop -A` suggests this alternative code, which is technically valid, but a little odd looking:

```ruby
def ===(other)
  @range === case other
             when IPRange
               other.range
             else
               other
             end
end
```

What strikes me as odd is no assignment is taking place inside the method, only a `===` comparison. I suspect that `rubocop` is confusing `===` for an assignment.

## Expected behavior

Not to flag `===` within `case` statements as a violation of `Style/ConditionalAssignment`.

## Actual behavior

```
For /home/postmodern/test/ruby/rubocop: Default configuration from /home/postmodern/test/ruby/rubocop/vendor/bundle/ruby/3.2.0/gems/rubocop-1.60.2/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /home/postmodern/.cache/rubocop_cache/5ad2eb8dfce914247e9e228bb447595b6b1fae8a/6d7a3b621ca1730e04accd938619e4bdab66cfb1/5132a40914f73434cc65cd035a9932f1eb6898e1
Inspecting 2 files
Scanning /home/postmodern/test/ruby/rubocop/Gemfile
Loading cache from /home/postmodern/.cache/rubocop_cache/5ad2eb8dfce914247e9e228bb447595b6b1fae8a/6d7a3b621ca1730e04accd938619e4bdab66cfb1/436b259af3311552bad97b22b2a96ea2817589da
.Scanning /home/postmodern/test/ruby/rubocop/test.rb
Loading cache from /home/postmodern/.cache/rubocop_cache/5ad2eb8dfce914247e9e228bb447595b6b1fae8a/6d7a3b621ca1730e04accd938619e4bdab66cfb1/5132a40914f73434cc65cd035a9932f1eb6898e1
C

Offenses:

test.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
def ===(other)
^
test.rb:3:3: C: [Correctable] Layout/EmptyLineAfterMagicComment: Add an empty line after magic comments.
  when IPRange
  ^
test.rb:2:3: C: [Correctable] Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
  case other ...
  ^^^^^^^^^^
test.rb:5:2: C: [Correctable] Layout/CaseIndentation: Indent when as deep as case.
  else
 ^^^^
test.rb:4:13: C: Style/CaseEquality: Avoid the use of the case equality operator ===.
    @range === other.range
            ^^^
test.rb:7:3: C: [Correctable] Layout/IndentationWidth: Use 2 (not -9) spaces for indentation.
  end ...
  ^^^
test.rb:9:7: C: [Correctable] Layout/ElseAlignment: Align else with when.
test.rb:9:16: C: [Correctable] Layout/IndentationWidth: Use 2 (not -9) spaces for indentation.

2 files inspected, 8 offenses detected, 7 offenses autocorrectable
Finished in 0.5618618270000297 seconds
```

## Steps to reproduce the problem

`Gemfile`:
```ruby
# frozen_string_literal: true

source 'https://rubygems.org'

gem 'rubocop'
```

`test.rb`:
```ruby
def ===(other)
  case other
  when IPRange
    @range === other.range
  else
    @range === other
  end
end
```

`bundle exec rubocop --debug`

## RuboCop version

```
$ bundle exec rubocop -V
1.60.2 (using Parser 3.3.0.5, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-linux]
```

**Comment by koic** [comment]
There may be some confusion regarding the cop name, but `Style/ConditionalAssignment` supports both variable assignment and variable comparison.

Changing the configuration from the default `EnforcedStyle: assign_to_condition` to `EnforcedStyle: assign_inside_condition` enables the desired behavior.

For more details, please refer to the documentation:
https://docs.rubocop.org/rubocop/1.60/cops_style.html#enforcedstyle-assign_inside_condition

**Comment by postmodern** [comment]
@koic changing to `EnforcedStyle: assign_inside_condition` causes rubocop to then flag all of the other actual assignments:

```ruby
foo = if condition
        1
        else
        2
```

I don't think `ConditionalAssignment` should include `===`, since no variable assignment is actually being done.
